### DISCUSSION TITLE
Contribution to pymoo with algorithms: MMOPSO

### DISCUSSION THREAD

**Discussion by UnderMou** [anyoptimization/pymoo#320] [Q&A]
Hello.

I'd like to know if has someone already contributed to pymoo framework by adding MMOPSO[1] multi-objective optimization algorithm to it. Nowadays I'm using NSGA-II, but I'll also have to compare the results of pareto front by other algorithms as well. Therefore, I want to keep using pymoo as it turned out to be a great tool for my previous analyses.

Before I dive into implementing these algorithms class in pymoo I'd like to know if someone is already engaged to it or even if it'd be feasable.
@blankjul would you have any recommendations/advices for implementing new algorithms to pymoo?
I'm already studying the documentation files

Thanks in advance.
Best regards.
AndersonMR.

References:

[1] Q. Lin, J. Li, Z. Du, J. Chen, and Z. Ming, A novel multi-objective particle swarm optimization with multiple search strategies, European Journal of Operational Research, 2015, 247(3): 732-744.

**Comment by blankjul** [comment]
My best advice is to look at existing implementations from other algorithms. I would probably pick an algorithm I already know.
I personally also like to simply debug and see all the steps.

I know Pymoo is written very modularly, which can be a little bit of a learning curve in the beginning.

Have you had some success? (I am asking because it already has been a while since you started the discussion)
